lib/matplotlib/axis.py

```diff
@@ -1515,37 +1515,101 @@ def set_major_formatter(self, formatter):
+        tick label.
+        See the  `~matplotlib.ticker.FuncFormatter` documentation for
+        more information.
+
         Parameters
         ----------
-        formatter : `~matplotlib.ticker.Formatter`
+        formatter : `~matplotlib.ticker.Formatter`, ``str``, or function
         """
-        cbook._check_isinstance(mticker.Formatter, formatter=formatter)
-        if (isinstance(formatter, mticker.FixedFormatter)
-                and len(formatter.seq) > 0
-                and not isinstance(self.major.locator, mticker.FixedLocator)):
-            cbook._warn_external('FixedFormatter should only be used together '
-                                 'with FixedLocator')
-        self.isDefault_majfmt = False
-        self.major.formatter = formatter
-        formatter.set_axis(self)
-        self.stale = True
+        self._set_formatter(formatter, self.major)
 
     def set_minor_formatter(self, formatter):
         """
         Set the formatter of the minor ticker.
 
+        In addition to a `~matplotlib.ticker.Formatter` instance,
+        this also accepts a ``str`` or function.
+
+        For a ``str`` a `~matplotlib.ticker.StrMethodFormatter` is used.
+        The field used for the value must be labeled ``'x'`` and the field used
+        for the position must be labeled ``'pos'``.
+        See the  `~matplotlib.ticker.StrMethodFormatter` documentation for
+        more information.
+
+        For a function a `~matplotlib.ticker.FuncFormatter` is used.
+        The function should take in two inputs (a tick value ``x`` and a
+        position ``pos``), and return a string containing the corresponding
+        tick label.
+        See the  `~matplotlib.ticker.FuncFormatter` documentation for
+        more information.
+
         Parameters
         ----------
-        formatter : `~matplotlib.ticker.Formatter`
+        formatter : `~matplotlib.ticker.Formatter`, ``str``, or function
+        """
+        self._set_formatter(formatter, self.minor)
+
+    def _set_formatter(self, formatter, level):
         """
-        cbook._check_isinstance(mticker.Formatter, formatter=formatter)
+        Set the formatter of the major or minor ticker.
+
+        In addition to a `~matplotlib.ticker.Formatter` instance,
+        this also accepts a ``str`` or function.
+
+        For a ``str`` a `~matplotlib.ticker.StrMethodFormatter` is used.
+        The field used for the value must be labeled ``'x'`` and the field used
+        for the position must be labeled ``'pos'``.
+        See the  `~matplotlib.ticker.StrMethodFormatter` documentation for
+        more information.
+
+        For a function a `~matplotlib.ticker.FuncFormatter` is used.
+        The function should take in two inputs (a tick value ``x`` and a
+        position ``pos``), and return a string containing the corresponding
+        tick label.
+        See the  `~matplotlib.ticker.FuncFormatter` documentation for
+        more information.
```

Comment:
Same as above, it should be in one canonical place with cross references. Maybe a short hand in comments for devs.

Reply:
Again, I was told I add more detail here.